core/templates/dev/head/pages/exploration_editor/ExplorationTitleEditor.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the exploration title field in forms.
+ */
+oppia.directive('explorationTitleEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope:{
+        // The text for the label Field.
+        labelName: '@',
+        // The value to be shown on focus.
+        onInputFieldFocus: '@',
+        // Additional CSS style to define the width and font-weight.
+        formStyle: '@',
+        // Additional CSS style for P tags.
+        pStyle:'@',
+        // The method to call on when the input field is blured.
+        onInputFieldBlur: '&',
+        // The value to determine the visibilty of the field.
```

Comment:
Hm. Can we not have this and just deal with visibility outside the component? It seems more like a job for the component's container.